solver/src/driver/solver_settlements.rs

```diff
@@ -1,79 +1,16 @@
 use crate::{encoding::EncodedSettlement, settlement::Settlement};
-use anyhow::Result;
 use ethcontract::U256;
 use num::BigRational;
 use primitive_types::H160;
 use shared::conversions::U256Ext;
 use std::{collections::HashMap, time::Duration};
-
-// Return None if the result is an error or there are no settlements remaining after removing
-// settlements with no trades.
-pub fn filter_bad_settlements(
-    (solver_name, solver_result): (&'static str, Result<Vec<Settlement>>),
-) -> Option<(&'static str, Vec<Settlement>)> {
-    let mut settlements = match solver_result {
-        Ok(settlements) => settlements,
-        Err(err) => {
-            tracing::error!("solver {} error: {:?}", solver_name, err);
-            return None;
-        }
-    };
-    settlements.retain(|settlement| !settlement.trades().is_empty());
-    if settlements.is_empty() {
-        return None;
-    }
-    Some((solver_name, settlements))
-}
-
-// A single solver produces multiple settlements
-#[derive(Debug)]
-pub struct SolverWithSettlements {
-    pub name: &'static str,
-    pub settlements: Vec<Settlement>,
-}
-
-impl From<SolverWithSettlements> for Vec<SettlementWithSolver> {
-    fn from(instance: SolverWithSettlements) -> Self {
-        let name = instance.name;
-        instance
-            .settlements
-            .into_iter()
-            .map(|settlement| SettlementWithSolver { name, settlement })
-            .collect()
-    }
-}
-
-#[derive(Debug, Clone)]
-pub struct SettlementWithSolver {
-    pub name: &'static str,
-    pub settlement: Settlement,
-}
-
-impl From<SettlementWithSolver> for EncodedSettlement {
-    fn from(instance: SettlementWithSolver) -> Self {
-        instance.settlement.into()
-    }
-}
-
-impl From<SettlementWithSolver> for Settlement {
-    fn from(instance: SettlementWithSolver) -> Self {
-        instance.settlement
-    }
-}
-
-impl SettlementWithSolver {
-    pub fn without_onchain_liquidity(&self) -> Self {
-        Self {
-            name: self.name,
-            settlement: self.settlement.without_onchain_liquidity(),
-        }
-    }
```

Comment:
Have we removed this completely or has it been replaced somewhere I can't see at the moment? I mean do we still have the option to settle orders with the contract internal buffers?

Reply:
Yes, I've just replaced `SettlementWithSolver` with `(SolverId, Settlement)`, so `SettlementWithSolver::without_onchain_liquidity` is now just `Settlement::without_onchain_liquidity`.

I'll add a unit test for this though.